includes/Settings.php

```diff
@@ -1008,6 +1012,213 @@ public function move_setting_after_id( $settings, $insert_settings, $after_setti
+
+					// Skip if the key has already been processed.
+					if ( in_array( $key, $processed_keys, true ) ) {
+						continue;
+					}
+
+					$settings_field             = $settings_fields[ $key ];
+					$settings_field['section']  = $category_name;
+					$modified_settings_fields[] = $settings_field;
+					$processed_keys[]           = $key;
+				}
+			}
+		}
+
+		// Check for any unprocessed settings fields.
+		$unprocessed_settings_fields = array_diff_key( $settings_fields, array_flip( $processed_keys ) );
+
+		// Create an "Additional settings" section for uncategorized settings fields.
+		if ( ! empty( $unprocessed_settings_fields ) ) {
+			$category_name              = 'additional';
+			$modified_settings_fields[] = array(
+				'type'     => 'section',
+				'id'       => $category_name,
+				'title'    => __( 'Additional settings', 'woocommerce-pdf-invoices-packing-slips' ),
+				'callback' => 'section',
+				'args'     => array(
+					'before_section' => '<div class="settings_category" id="' . $category_name . '">',
+					'after_section'  => '</div>',
+				),
+			);
+
+			// Add rest of settings to the $modified_settings_fields array under "More" category
+			foreach ( $unprocessed_settings_fields as $settings_field ) {
+				$settings_field['section']  = $category_name;
+				$modified_settings_fields[] = $settings_field;
+			}
+		}
+
+		return $modified_settings_fields;
+	}
+
+	/**
+	 * Helper method to add a single setting field to a category.
+	 *
+	 * @param array           $settings_categories
+	 * @param string          $new_setting_id
+	 * @param string          $category_name
+	 * @param string|int|null $position
+	 *
+	 * @return array
+	 */
+	public function add_single_setting_field_to_category( array $settings_categories, string $new_setting_id, string $category_name, $position = null ): array {
+		return $this->add_setting_field_to_category_impl( $settings_categories, [ $new_setting_id ], $category_name, $position );
+	}
+
+	/**
+	 * Helper method to add multiple setting fields to a category.
+	 *
+	 * @param array           $settings_categories
+	 * @param array           $new_setting_ids
+	 * @param string          $category_name
+	 * @param string|int|null $position
+	 *
+	 * @return array
+	 */
+	public function add_multiple_setting_fields_to_category( array $settings_categories, array $new_setting_ids, string $category_name, $position = null ): array {
+		return $this->add_setting_field_to_category_impl( $settings_categories, $new_setting_ids, $category_name, $position );
+	}
+
+	/**
+	 * Internal method to handle adding setting fields to a category.
+	 *
+	 * @param array           $settings_categories
+	 * @param array           $new_setting_ids
+	 * @param string          $category_name
+	 * @param string|int|null $position
+	 *
+	 * @return array
+	 */
+	private function add_setting_field_to_category_impl( array $settings_categories, array $new_setting_ids, string $category_name, $position = null ): array {
```

Comment:
Maybe `impl` is not needed?

Reply:
This was actually a mistake when I was merging the main; thank you.